### PR TITLE
Add support for WebAuthn (leaving U2F support in place)

### PR DESCRIPTION
**Pull request by forevermatt**
### Added
- Use npm to install client library for handling WebAuthn
- Add template (etc.) files for WebAuthn support

### Fixed
- Explicitly require the JSON PHP extension, since we use it
- Update the copyright year in the license file

### CODE REVIEW

File: themes/material/mfa/prompt-for-mfa-webauthn.php [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+            var errorNode = document.querySelector('p.error');
+
+            errorNode.classList.remove('hide');
+            errorNode.querySelector('span').textContent = message;
+
+            offerRetry();
+        }
+
+        /**
+         * https://developers.yubico.com/U2F/Libraries/Client_error_codes.html
```
Comment (longrunningprocess) [inline themes/material/mfa/prompt-for-mfa-webauthn.php]: not sure whether this comment will still be accurate or not...

**Review by longrunningprocess — COMMENTED** [review]
looks pretty thorough to me, nothing I'd hold the PR up for so far.  Excellent work as always 🚀

File: composer.json [hunk 2]
```diff
@@ -12,6 +12,7 @@
   "minimum-stability": "stable",
   "require": {
     "php": ">=7.0",
+    "ext-json": "*",
```
[the remark this answers is not in the document]
Comment (forevermatt) [reply composer.json:15]: The advantages of a good IDE... 🙂

File: dictionaries/mfa.definition.json [hunk 3]
```diff
@@ -102,6 +132,18 @@
 		"fr": "Cela a pris un peu trop de temps, vérifiez que votre clé est insérée dans le bons sens.",
 		"ko": "오랜 시간이 경과 되었으니 키가 오른쪽 위로 삽입되었는지 확인 하십시오."
 	},
+	"webauthn_error_abort": {
+		"en": "It looks like you clicked cancel. Would you like us to try again?",
+		"es": "",
+		"fr": "",
```
Comment (forevermatt) [inline dictionaries/mfa.definition.json:138]: @Baggerone, could you add a French translation here and in the block below?

File: themes/material/mfa/prompt-for-mfa-webauthn.php [hunk 4]
```diff
@@ -0,0 +1,145 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <title><?= $this->t('{material:mfa:title}') ?></title>
+
+    <?php include __DIR__ . '/../common-head-elements.php' ?>
+
+    <?php
+    $webauthnJsFileHash = md5_file(__DIR__ . '/../../../www/simplewebauthn/browser.js');
```
[the remark this answers is not in the document]
Comment (forevermatt) [reply themes/material/mfa/prompt-for-mfa-webauthn.php:9]: Thanks 🙂

File: dictionaries/mfa.definition.json (hunk 3 above)
Comment (Baggerone) [reply dictionaries/mfa.definition.json]: Just pushed them.

**Review by fillup — APPROVED** [review]

**Pull request merged**